packages/core/lib/commands/compile.js

```diff
@@ -94,28 +92,60 @@ const command = {
   },
 
   listVersions: async function (options) {
-    const {CompilerSupplier} = require("@truffle/compile-solidity");
+    const { CompilerSupplier } = require("@truffle/compile-solidity");
+    const { asyncTake } = require("iter-tools");
+
     const supplier = new CompilerSupplier({
-      solcConfig: options.compilers.solc,
+      solcConfig: {
+        ...options.compilers.solc,
+        // HACK to force use of the VersionRange or Docker strategy
+        // as implemented, Docker requires a version to be specified, and so
+        // we can't simply remove this field entirely.
+        version: "0.5.16",
+        docker: options.list === "docker"
+      },
       events: options.events
     });
 
     const log = options.logger.log;
     options.list = options.list.length ? options.list : "releases";
 
-    // Docker tags
-    if (options.list === "docker") {
-      const tags = await supplier.getDockerTags();
-      tags.push("See more at: hub.docker.com/r/ethereum/solc/tags/");
-      log(format(tags, null, " "));
+    const {
+      latestRelease,
+      releases,
+      prereleases
+    } = await supplier.list();
```

Comment:
@haltman-at asks about Local/Native strategies [here](https://github.com/trufflesuite/truffle/pull/4222#discussion_r687283884)

Reply:
Turns out we already addressed this in #4222